samza-api/src/main/java/org/apache/samza/operators/windows/Windows.java

```diff
@@ -114,10 +117,10 @@ private Windows() { }
    * @return the created {@link Window} function.
    */
   public static <M, K, WV> Window<M, K, WV>
-    keyedTumblingWindow(Function<M, K> keyFn, Duration interval, BiFunction<M, WV, WV> foldFn) {
+    keyedTumblingWindow(Function<M, K> keyFn, Duration interval, Supplier<WV> initialValue, BiFunction<M, WV, WV> foldFn) {
```

Comment:
IIUC, aggregator, key Function, initialValue supplier etc. are purely stateless JDK style `Function` classes. If there's state needed, can't we always access it from the context manager? A nice property is that users can use JDK style functional interfaces in code.